korman/operators/op_mesh.py

```diff
@@ -24,6 +24,109 @@ def poll(cls, context):
         return context.scene.render.engine == "PLASMA_GAME"
 
 
+class PlasmaAddFlareOperator(PlasmaMeshOperator, bpy.types.Operator):
+    bl_idname = "mesh.plasma_flare_add"
+    bl_label = "Add Lamp Flare"
+    bl_category = "Plasma"
+    bl_description = "Adds a new Plasma Lamp Flare"
+    bl_options = {"REGISTER", "UNDO"}
+
+    # Allows user to specify their own name stem
+    flare_name = bpy.props.StringProperty(name="Name",
+                                         description="Flare name stem",
+                                         default="Flare",
+                                         options=set())
+    flare_distance = bpy.props.FloatProperty(name="Distance",
+                                             description="Flare's distance from the illuminating object",
+                                             min=0.1, max=2.0, step=10, precision=1, default=1.0,
+                                             options=set())
+
+    def draw(self, context):
+        layout = self.layout
+        space = bpy.context.space_data
+
+        if not space.local_view:
+            box = layout.box()
+            box.label("Flare Name:")
+            row = box.row()
+            row.alert = not self.flare_name
+            row.prop(self, "flare_name", text="")
+
+            box = layout.box()
+            box.label("Geometry:")
+            row = box.row()
+            row.prop(self, "flare_distance")
+        else:
+            row = layout.row()
+            row.label("Warning: Operator does not work in local view mode", icon="ERROR")
+
+    def execute(self, context):
+        if context.mode == "OBJECT":
+            self.create_flare_objects()
+        else:
+            self.report({"WARNING"}, "Lamp Flare creation only valid in Object mode")
```

Comment:
Might be better to adjust the `poll()` to check for object mode.

Reply:
👍